L1Trigger/L1TMuonOverlapPhase2/interface/LutNeuronLayerFixedPoint.h

```diff
@@ -0,0 +1,190 @@
+//============================================================================
+// Name        : LutNeuronLayerFixedPoint.h
+// Author      : Karol Bunkowski
+// Created on: Mar 12, 2021
+// Version     :
+// Copyright   : All right reserved
+// Description : Fixed point LUT layer
+//============================================================================
+
+#ifndef INTERFACE_LUTNEURONLAYERFIXEDPOINT_H_
+#define INTERFACE_LUTNEURONLAYERFIXEDPOINT_H_
+
+#include <ap_fixed.h>
+#include <ap_int.h>
+#include <array>
+#include <limits>
+#include <iomanip>
+#include <cassert>
+
+#include <boost/property_tree/ptree.hpp>
+
+#include "L1Trigger/L1TMuonOverlapPhase2/interface/LutNetworkFixedPointCommon.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+namespace lutNN {
+  // constexpr for ceil(log2) from stackoverflow
+  constexpr size_t floorlog2(size_t i) {
+    assert(i > 0);  // argument of floorlog2 must be grater than 0
```

Comment:
For single asserts, it may be better to throw a `cms::Exception`, which may allow a slightly more verbose explained version of what happened, rather than pointing out a line number and a failure.